classes/class-query.php

```diff
@@ -118,12 +118,10 @@ public function query( $args ) {
 
 				$field = str_replace( array( 'record_', '__in' ), '', $key );
 				$field = empty( $field ) ? 'ID' : $field;
-				$type  = is_numeric( array_shift( $value ) ) ? '%d' : '%s';
+				$type  = is_numeric( $value[ 0 ] ) ? '%d' : '%s';
```

Comment:
This won't work if the first value in the array isn't `0`. Could we use this as a non-mutating solution, instead?
`array_shift( array_values( $value ) );`

Reply:
But in what case wouldn't the first item's array index be exactly 0? Can there be string array index values, or?

I've yet to see any PHP warnings or anything in the 1 year this has been running in production. Wouldn't there be?